Demo/Demo/Features/FeatureBaseViewController.swift

```diff
@@ -0,0 +1,91 @@
+        NSLayoutConstraint.activate([
+            containerView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
+            containerView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
+            containerView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
+            containerView.heightAnchor.constraint(equalToConstant: 60),
+
+            titleLabel.leadingAnchor.constraint(equalTo: containerView.leadingAnchor),
+            titleLabel.trailingAnchor.constraint(equalTo: containerView.trailingAnchor),
+            titleLabel.centerYAnchor.constraint(equalTo: containerView.centerYAnchor)
+        ])
+    }
+
+    func fetchOrderID() {
+        // TODO: The prod server is broken right now, so we will expect those requests to fail.
+        // TODO: Pass order details into this function
+
+        let amount = Amount(currencyCode: "USD", value: "10.00")
+        let orderRequestParams = CreateOrderParams(
+            intent: DemoSettings.intent.rawValue.uppercased(),
+            purchaseUnits: [PurchaseUnit(amount: amount)]
+        )
+
+        DemoMerchantAPI.sharedService.createOrder(orderParams: orderRequestParams) { result in
+            self.processOrder(status: "Processing your order")
+            switch result {
+            case .success(let order):
+                self.processOrder(status: "Successfully \(order.status.lowercased()) your order")
+                print("✅ fetched orderID: \(order.id) with status: \(order.status)")
+                DispatchQueue.main.async {
+                    self.displayAlert(message: "Order ID: \(order.id) has a status of \(order.status.lowercased())", success: true)
+                }
+            case .failure(let error):
+                self.processOrder(status: "Your order has failed, please try again")
+                print("❌ failed to fetch orderID: \(error)")
+                DispatchQueue.main.async {
+                    self.displayAlert(message: "Error: \(error)", success: false)
+                }
+            }
+        }
+    }
+
+    // TODO: Do we need this/the bottom container or are we fine using the alert function instead to display these details?
+    func processOrder(status: String) {
+        DispatchQueue.main.async {
+            self.titleLabel.text = status
+        }
+    }
+
+    private func displayAlert(message: String, success: Bool) {
+        let successTitle = "😎 Successful Order"
+        let failureTitle = "😕 Failed Order"
+        let title = success ? successTitle : failureTitle
+
+        let alertController = UIAlertController(title: title, message: message, preferredStyle: .alert)
+        let okayAction = UIAlertAction(title: "Okay", style: .default) { _ in
+            self.navigationController?.popViewController(animated: true)
+        }
+
+        alertController.addAction(okayAction)
+        present(alertController, animated: true, completion: nil)
+    }
+}
```

Comment:
So for this base/parent class that all of the `<PaymentMethodType>Feature` classes will inherit from, I think we'll want it to do a few things, which will be relevant to all payment methods:
- Fetch an orderID
- Have some sort of status/view label that displays the state of the flow
- Have a button that allows us to `process` order (capture/authorize) after the flow completes

I'm not sure an alert is the most helpful here, since we will likely want to show multiple types of status updates such as "orderID fetched: \(...)", "cardApproval in process" etc. We also probably want a button nearby this status bar for "Process Order" since that is something we'll want to test for each payment flow.

Let me know if this makes sense!

Reply:
I think that makes sense. Let me think on this a little more and get an idea implemented. I think initially, I liked the idea of some details (like success/failure) being in an alert to keep the bottom status bar from getting to busy, but I also understand what you're saying!

Reply:
Curious to hear your thoughts on the updated implementation of this @scannillo! Maybe we can sync up and chat about it if I'm still not quite following. 😄

Reply:
Nope, this looks good now!